File: TCS/NQT/NQT_1.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

void bruteForce(vector<int>& arr, int n, int target) {
    for (int i = 0; i < n; i++) {
        int curSum = 0;
        for (int j = i; j < n; j++) {
            curSum += arr[j];
            if (curSum == target) {
                for (int k = i; k <= j; k++) {
                    cout << arr[k] << " ";
                }
                cout << endl;
            }
        }
    }
}


int main() {
    vector<int> arr = {3, 4, -7, 1, 3, 3, 1, -4};
    int N = arr.size();
    int target = 7;
    bruteForce(arr, N, target);
    return 0;
}

File: TCS/NQT/NQT_1.py
def brute_force(arr, target):
    n = len(arr)
    for i in range(n):
        cur_sum = 0 # current sum
        for j in range(i, n): # iterate from i to n
            cur_sum += arr[j] #arr[j] means the jth element of the array
            if cur_sum == target:
                print(" ".join(map(str, arr[i:j+1]))) #print the elements from i to j. j+1 means the jth element is not included. if it was j then the jth element would be included.

if __name__ == "__main__":
    arr = list(map(int, input("Enter the array elements separated by spaces: ").split()))
    target = int(input("Enter the target sum: "))
    brute_force(arr, target)
#the logic of this approach is to iterate through the array and check for the sum of the elements from i to j. If the sum is equal to the target sum, then print the elements from i to j. time complexity of this approach is O(n^2). space complexity is O(1).